src/vm/docs/mod.rs

```diff
@@ -271,6 +271,46 @@ Note: This function is only available starting with Stacks 2.1.",
 "#,
 };
 
+const PRINCIPAL_PARSE_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(principal-parse principal-address)",
+    description: "A principal value is a concatenation of two components: a single *version byte*
+(indicating the type of account and the type of network that this principal can spend tokens on),
+and a 20-byte *public key hash* (indicating the principal's unique identity).
+`principal-parse` will decompose a principal into its component parts, returning the pair as a tuple.
+
+This method will return an `InvalidVersionByte` error if the input `principal-address` does
+not begin with a prefix that corresponds to a valid version byte (see `principal-construct`).
+
+Note: This function is only available starting with Stacks 2.1.",
+    example: r#"
+(principal-parse 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; Returns (tuple (hash-bytes 0x164247d6f2b425ac5771423ae6c80c754f7172b0) (version 0x1a))
+"#,
+};
+
+const PRINCIPAL_CONSTRUCT_API: SimpleFunctionAPI = SimpleFunctionAPI {
+    name: None,
+    signature: "(principal-construct version-byte hash-bytes)",
+    description: "A principal value is a concatenation of two things: a 1-byte *version byte*,
+indicating the type of account and the type of network that this principal can spend tokens on,
+and a 20-byte *public key hash*, characterizing the principal's unique identity.
+`principal-construct` takes such a `version-byte`, of type `(buff 1)`, 
+and `hash-bytes`, of type `(buff 20)`, and creates a corresponding object of type `principal`.
+
+The version byte should be `0x16` for a single-signature account on mainnet, `0x14`
+for a multi-signature account on mainnet, `0x1a` for a single-signature account on
+a testnet, `0x15` for a multi-signature account on a testnet. The public key hash
+should be a 20-byte buffer containing the hash of a public key.
+
+This method will return an `InvalidVersionByte` error if the input `version-byte` does
```

Comment:
I really don't think this function should outright fail if the user gives bad input.  Like other functions that can fail in this way, this function could easily return a `(response)` type that tells the caller why it won't work.  If the caller wants the transaction to abort or short-circuit, they can use `unwrap-panic` or `try!`.

Reply:
I had it as not failing initially. Then I think @lgalabru and @pavitthrap wanted it to fail for bad input.

Returning a `response` is a separate question. I can implement either but we should get some kind of consensus or at least decision as to what to do.